src/group_theory/submonoid/membership.lean

```diff
@@ -233,12 +233,18 @@ lemma pow_right_injective_iff_pow_injective {n : M} :
   function.injective (λ m : ℕ, n ^ m) ↔ function.injective (pow n) :=
 subtype.coe_injective.of_comp_iff (pow n)
 
-theorem log_pow_eq_self [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)) (m : ℕ) :
-  log (pow n m) = m :=
-pow_right_injective_iff_pow_injective.mp h $ pow_log_eq_self _
+/-- The exponentiation map is an isomorphism from the additive monoid on natural numbers to powers
+when it is injective. The inverse is given by the logarithms. -/
+def pow_log_equiv [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)):
+  multiplicative ℕ ≃* powers n :=
+{ to_fun := pow n,
```

Comment:
I think you could also prove this using `mul_equiv.of_left_inverse` which would probably be defeq and maybe shorter. Also fine as is

Reply:
I see, thanks! I'll leave it like this for now just coz it's very explicit and all one-liner.